tests/TestCase/Datasource/Paging/PaginatorTestTrait.php

```diff
@@ -1167,13 +1169,10 @@ public function testPaginateQueryUnusedFinder(): void
         $table->expects($this->never())
             ->method('find');
 
-        $query->expects($this->once())->method('applyOptions')
-            ->with([
-                'limit' => 2,
-                'page' => 1,
-                'order' => [],
-            ]);
-        $this->Paginator->paginate($query, [], $settings)->pagingParams();
```

Comment:
We convert warnings to exceptions, therefore we can't expect methods to be called if we properly expect the warning message (since this method isn't actually called anymore)

This of course doesn't really test the `applyOptions` call anymore, but at least we don't have any warnings.